src/main/java/org/terasology/gookeeper/system/EconomySystem.java

```diff
@@ -177,63 +178,69 @@ public String upgrade(@Sender EntityRef client,
         return "Couldn't find the requested item to upgrade.";
     }
 
-    private void upgradeByType (EntityRef item, UpgradableComponent upgradableComponent) {
-        upgradableComponent.currentTier ++;
+    private void upgradeByType(EntityRef item, UpgradableComponent upgradableComponent) {
+        upgradableComponent.currentTier++;
 
         if (item.hasComponent(SlimePodItemComponent.class)) {
             SlimePodItemComponent slimePodItemComponent = item.getComponent(SlimePodItemComponent.class);
-            slimePodItemComponent.slimePods = upgradableComponent.baseQuantity * (1 + upgradableComponent.currentTier * upgradableComponent.baseQuantityMultiplier);
+            slimePodItemComponent.slimePods =
+                    upgradableComponent.baseQuantity * (1 + upgradableComponent.currentTier * upgradableComponent.baseQuantityMultiplier);
             item.saveComponent(slimePodItemComponent);
         } else if (item.hasComponent(PlazMasterComponent.class)) {
             PlazMasterComponent plazMasterComponent = item.getComponent(PlazMasterComponent.class);
-            plazMasterComponent.charges = upgradableComponent.baseQuantity * (1 + upgradableComponent.currentTier * upgradableComponent.baseQuantityMultiplier);
+            plazMasterComponent.charges =
+                    upgradableComponent.baseQuantity * (1 + upgradableComponent.currentTier * upgradableComponent.baseQuantityMultiplier);
             plazMasterComponent.maxCharges = plazMasterComponent.charges;
             item.saveComponent(plazMasterComponent);
         }
     }
 
     /**
-     * This function is to be called by a visitor entity when it gets spawned into the world
-     * Adds up credit in the player wallet in form of an entrance fee.
+     * This function is to be called by a visitor entity when it gets spawned into the world Adds up credit in the
+     * player wallet in form of an entrance fee.
      *
-     * @param visitor The visitor entity
+     * @param visitorComponent The visitor entity's {@link VisitorComponent}
      */
     @Override
-    public void payEntranceFee (EntityRef visitor) {
-        VisitorComponent visitorComponent = visitor.getComponent(VisitorComponent.class);
-        VisitorEntranceComponent visitorEntranceComponent = visitorComponent.visitorEntranceBlock.getComponent(VisitorEntranceComponent.class);
-        EntityRef player = visitorEntranceComponent.owner;
-        EconomyComponent economyComponent = player.getComponent(EconomyComponent.class);
-
-        if (economyComponent != null) {
-            economyComponent.playerWalletCredit += baseEntranceFee;
-            player.saveComponent(economyComponent);
+    public void payEntranceFee(VisitorComponent visitorComponent) {
+        if (visitorComponent.visitorEntranceBlock.hasComponent(VisitorEntranceComponent.class)) {
```

Comment:
Should we log something in case the component is not there?

Reply:
good idea, I'll take care of it and re-request review afterwards 👍